TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/RoutesToProfessionalStatus/EditRoute/InductionExemption.cshtml.cs

```diff
@@ -68,27 +72,14 @@ public async Task<IActionResult> OnPostCancelAsync()
         return Redirect(linkGenerator.PersonQualifications(PersonId));
     }
 
-    public async override Task OnPageHandlerExecutionAsync(PageHandlerExecutingContext context, PageHandlerExecutionDelegate next)
+    public override Task OnPageHandlerExecutionAsync(PageHandlerExecutingContext context, PageHandlerExecutionDelegate next)
     {
-        Route = await referenceDataCache.GetRouteToProfessionalStatusByIdAsync(JourneyInstance!.State.RouteToProfessionalStatusId);
-        if (Route.InductionExemptionReasonId is not null)
-        {
-            var exemptionReason = await referenceDataCache.GetInductionExemptionReasonByIdAsync(Route.InductionExemptionReasonId!.Value);
-            if (exemptionReason.RouteImplicitExemption)
-            {
-                context.Result = new BadRequestResult();
-            }
-        }
-        else
-        {
-            context.Result = new BadRequestResult();
-        }
-
         var personInfo = context.HttpContext.GetCurrentPersonFeature();
         PersonName = personInfo.Name;
         PersonId = personInfo.PersonId;
-
-        await base.OnPageHandlerExecutionAsync(context, next);
+        var professionalStatusFeature = context.HttpContext.GetCurrentProfessionalStatusFeature();
+        Route = professionalStatusFeature!.ProfessionalStatus.Route;
+        return base.OnPageHandlerExecutionAsync(context, next);
```

Comment:
this is the change to edit that i should have done with that migration and EF navigation to make it easier